platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/lag/BenchmarkSlowAsyncOutputStream.java

```diff
@@ -58,11 +62,17 @@ public void sendAsync(final T message) throws InterruptedException {
      */
     @Override
     protected void serializeMessage(final T message) throws IOException {
-        try {
-            Thread.sleep(delayMilliseconds);
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-        }
+        sleepMicros(delayNetworkMicroseconds);
         message.serialize(getOutputStream());
     }
+
+    /**
+     * Busy sleep for a given number of microseconds.
+     * @param micros time to sleep, in microseconds
+     */
+    private static void sleepMicros(final long micros) {
+        final long sleepUntil = System.nanoTime() + micros * 1000L;
+        while (System.nanoTime() < sleepUntil)
```

Comment:
There's no need for a spinwait here.
Thread has a static [sleep](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Thread.html#sleep(java.time.Duration)) method with sufficient precision.
```java
Thread.sleep(Duration.ofNanos(micros * 1000));
```

Reply:
Ah, it's available since Java 19. Good to learn about it. Let me update the fix.

Reply:
Note that it's not as precise as the busy wait. And it in fact slows down the benchmark by some 18%. But I guess it's still better than a busy loop. So the latest commit implements this.